wallets/core/src/wallet.ts

```diff
@@ -76,7 +76,11 @@ class Wallet<InstanceType = any> {
     return await this.connect(network);
   }
 
-  async connect(network?: Network, namespaces?: Namespace[]) {
+  async connect(
+    network?: Network,
+    namespaces?: Namespace[],
+    extra?: { derivationPath?: string }
```

Comment:
Please define a type called `ConnectExtraValues` (you can rename it to a better name), use it here and also change `wallets/core/src/types.ts:64` from `derivationPath?: string;` to `extra: ConnectExtraValues` to pass whole object at once.

Reply:
Fixed

Reply:
Is there any reason to define `ConnectExtraParams` in `wallets/react/src/types.ts` and `wallets/shared/src/rango.ts`?

Reply:
Removed from `wallets/react/src/types.ts`